tests/data_group_test.py

```diff
@@ -252,12 +252,20 @@ def test_getitem_positional_indexing_without_dim_label_raises_unless_1d():
         dg2d[::]
 
 
-def test_getitem_positional_indexing_raises_when_length_is_not_unique():
+def test_getitem_positional_slice_does_not_raise_when_length_is_not_unique():
     dg = sc.DataGroup({'a': sc.arange('x', 4), 'b': sc.arange('x', 5)})
-    with pytest.raises(sc.DimensionError):
+    dg['x', 2:3]
+    dg['x', 2:]
+    dg['x', 2:10]
+    dg['x', 2:4]
+    dg['x', -1:]
+    dg['x', -10:]
+
+
+def test_getitem_positional_indexing_raise_when_one_is_empty():
+    dg = sc.DataGroup({'a': sc.array(dims='x', values=[]), 'b': sc.arange('x', 5)})
+    with pytest.raises(IndexError):
         dg['x', 2]
```

Comment:
Can we get tests for the non-failure cases (mixed length), and failure case (mixed length but not zero)?